src/Native/Runtime/unix/UnwindHelpers.cpp

```diff
@@ -496,7 +729,7 @@ UnwindInfoSections LocateUnwindSections(uintptr_t pc)
     }
 #else // __APPLE__
 
-#if _LIBUNWIND_SUPPORT_DWARF_UNWIND
+#if _LIBUNWIND_SUPPORT_DWARF_UNWIND && !defined(_TARGET_ARM_) && !defined(_TARGET_ARM64_)
```

Comment:
Is `!defined(_TARGET_ARM_)` here intentional? I believe people have succeeded building this for ARM before.

Reply:
I am getting this when _TARGET_ARM_ is not excluded, even though I am solely targeting ARM64:
```
/corert/src/Native/Runtime/unix/UnwindHelpers.cpp:733:5: error: use of undeclared identifier 'dl_iterate_cb_data'; did you mean 'dl_iterate_phdr'?
```

Maybe _TARGET_ARM_ is being set for ARM64 too somewhere? I never figured out why it was behaving like that,

Reply:
I just checked again and it looks like `_LIBUNWIND_SUPPORT_DWARF_INDEX` is not being set while `_LIBUNWIND_SUPPORT_DWARF_UNWIND` is.

Reply:
Looks like this has already been fixed at LLVM's libunwind repository: https://github.com/llvm-mirror/libunwind/blob/master/src/AddressSpace.hpp#L473

Reply:
> Is there any particular procedure to update libunwind on CoreRT?

https://github.com/dotnet/corert/pull/7525

Reply:
libunwind is updated now

Reply:
I am getting a myriad of errors with the new version of libunwind (mostly method signatures that have changed). I'll update the discussion when I get them sorted out.

Reply:
Did the version of libunwind the repository was updated to compile on its own?
There was a missing ";" at `/corert/src/Native/Runtime/../libunwind/src/Registers.hpp` line 1966 and I am also getting this error:
```
/corert/src/Native/Runtime/../libunwind/src/Registers.hpp:1963:3: error: static_assert failed "arm64 registers do not fit into unw_context_t"
  static_assert((check_fit<Registers_arm64, unw_context_t>::does_fit),
  ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I'm not sure how to solve this, it doesn't seem to be an issue in CoreRT at first glance but rather a problem in libunwind itself.

Reply:
We have a CoreRT specific changes on top of the stock libunwind.

The missing ";" was my mistake in resolving merge conflicts.

> static_assert((check_fit<Registers_arm64, unw_context_t>::does_fit),

This one should be fixed by increasing the value of ARM64-specific _LIBUNWIND_CONTEXT_SIZE in `__libunwind_config.h`. I think it should be increased to 100

```
<<< #  define _LIBUNWIND_CONTEXT_SIZE 66
>>> #  define _LIBUNWIND_CONTEXT_SIZE 100
```

Reply:
That makes sense. 
Is the lack of `uint64_t location` parameters on some of the Registers_arm64 method definitions also due to this? The parameter is present on the definition of the body, and I get no compile errors if I add the parameter back to the definition.
Example:
`void        setRegister(int num, uint64_t value);`
To:
`void        setRegister(int num, uint64_t value, uint64_t location);`

I guess _LIBUNWIND_CURSOR_SIZE will also require an increase in its value since I am now getting `static_assert failed "UnwindCursor<> does not fit in unw_cursor_t"`. I am not sure what value it should be set, however.

Reply:
You are right on both counts.

Reply:
Never mind the last part about the cursor, adding `34` to the cursor size made the error go away.